### ISSUE TITLE
Request for separable_convolution2d or depthwise_separable_conv originally from MobileNet or Xception

### ISSUE THREAD

**Issue by sharpstill** [ppwwyyxx/tensorpack#579]
Request feature layer : `separable_convolution2d ` or `group_conv`(in paper ShuffleNet) or `depthwise_separable_conv `(Xception or MobileNet) 
As far as I know, tensorflow slim has `slim.separable_convolution2d ` but tensorpack still not support it?

**Comment by ppwwyyxx** [comment]
You can call slim or `tf.layers` in tensorpack.
As mentioned in the README, tensorpack is NOT a model wrapper.

Duplicate of #236. Closing.

**Comment by ppwwyyxx** [comment]
Also easy to implement by yourself (about 10 lines of code), e.g. in shufflenet:
https://github.com/ppwwyyxx/tensorpack/blob/c266152797221365d5f9b7377c99423b8c1a1aca/examples/ShuffleNet/shufflenet.py#L29-L44

**Comment by sharpstill** [comment]
Thanks a lot , your example code sames also support group conv original from paper ShuffleNet(Face++), I just found the tf.nn support it, the trick is you just pass the filter_shape to W.
Without you , I cannot get the point and image how to implement it yet, can you put the example tutorial in Your Official Document?

**Comment by ppwwyyxx** [comment]
I don't know what you mean by "the example tutorial". All examples are in https://github.com/ppwwyyxx/tensorpack/tree/master/examples/ as you can see.

**Comment by sharpstill** [comment]
one question please, I see the tf document about` tf.nn.depthwise_conv2d`,
which is located in https://www.tensorflow.org/api_docs/python/tf/nn/depthwise_conv2d
 What is filter's last  dimension `channel_multiplier`, I cannot understand why the doment says output has **in_channels * channel_multiplier** channels.
If I am not understand wrongly, the ShuffleNet paper's `group convolution ` is output channel is equals with the input channel????
Does the ShuffleNet paper's `group convolution  ` mean each input feature map's channel will be convolved by multiple kernel channel, and this will cause the output channel is **in_channels * channel_multiplier** ?

**Comment by ppwwyyxx** [comment]
You can set channel_multiplier to 1.

**Comment by sharpstill** [comment]
I just cannot get the ShuffleNet paper's  ` group convolution ` output channel == input channel? in your example, it seems not like this?  I read the paper, and did not found the useful information about how to does `group convolve` exactly?
Can you write just some information to help me please?

**Comment by ppwwyyxx** [comment]
>  the doment says output has in_channels * channel_multiplier channels

So if you set channel_multiplier to 1, output channel == input channel.

**Comment by sharpstill** [comment]
In your  models.Conv2D: there is split parameter:
split (int): Split channels as used in Alexnet. Defaults to 1 (no split).
What does this mean, does this actually do **1x1 group conv** in ShuffleNet?

**Comment by sharpstill** [comment]
Thanks a lot, I read source code, split does actually a lot of individual conv layer here.

**Comment by liuhu-bigeye** [comment]
One question, note that TF r1.5 support cudnn 7.0 which highlighted with a faster group convolution implementation, will tensorpack add group conv based on cudnn 7.0 instead?

**Comment by ppwwyyxx** [comment]
AFAIK tensorflow still does not use cudnn7 group conv.
